reducers/plugins/index.js

```diff
@@ -136,10 +168,18 @@ function components(state = {}, action) {
     case ActionTypes.RECEIVED_PLUGIN_COMPONENT: {
         if (action.name && action.data) {
             const nextState = {...state};
-            const currentArray = nextState[action.name] || [];
-            const nextArray = [...currentArray];
+            const currentArray = (nextState[action.name] || []).filter((c) => !c.subMenu);
```

Comment:
wouldn't `const currentArray = nextState[action.name] || [];` just suffice here?

Reply:
with current logic, without this, we have subtrees that aren't attached to a parent.  
<img width="389" alt="Screen Shot 2019-10-05 at 12 02 12 PM" src="https://user-images.githubusercontent.com/2914443/66259631-11a2ae80-e768-11e9-93f6-14ee3e7d386e.png">

Reply:
i see, but this is so if you append `action.data` directly to `currentArray`, right? What if you were to update the menus inline somewhere else, that way you don't have to filter out the `c.subMenu`?

Reply:
ok, updated with the following.  
```javascript
            let currentArray = nextState[action.name] || [];
            if (action.name === 'PostDropdownMenu' && actionData.parentMenuId) {
                const subMenus = (nextState[action.name] || []).filter((c) => c.subMenu && c.pluginId === actionData.pluginId);
                const subMenu = subMenus.find((sm) => hasSubmenu(sm, actionData));
                if (!subMenu) {
                    return state;
                }
                currentArray = currentArray.filter(ca => ca.id !== subMenu.id);
                actionData = buildSubMenu(subMenu, actionData);
            }
            const nextArray = [...currentArray, actionData];
```

currentArray is initialized as requested, and updated/filtered if a subMenu was found.